stdlib/number.gr

```diff
@@ -670,3 +670,84 @@ provide let toRadians = degrees => degrees * (pi / 180)
  * @since v0.5.4
  */
 provide let toDegrees = radians => radians * (180 / pi)
+
+/**
+ * Constrains a number to a given range (inclusive).
+ *
+ * @param range: The inclusive range to clamp within.
+ * @param input: The number to clamp
+ * @returns The constrained number
+ *
+ * @since v0.6.0
+ */
+provide let clamp = (range, input) => {
+  if (isNaN(input)) {
+    input
+  } else {
+    let rangeEnd = max(range.rangeStart, range.rangeEnd)
+    let rangeStart = min(range.rangeStart, range.rangeEnd)
+
+    if (input > rangeEnd) rangeEnd else if (input < rangeStart) rangeStart
+    else input
+  }
+}
+
+/**
+ * Blends between a linear range (inclusive) of numbers using a given weight.
```

Comment:
```suggestion
 * Finds a value within a linear inclusive range of numbers using a given weight.
```
Thoughts on this description? Seems more descriptive than "blends"

Reply:
The only real issue i see with that. is i dont think it really describes how the weight is used. compared to blends where you can kind of see that it is merging the two.